File: NSPersistentContainer/NSPersistentContainerObjCSample/NSPersistentContainerObjCSample/SampleTableViewController.m
//
//  SampleTableViewController.m
//  NSPersistentContainerObjCSample
//
//  Created by nikolay.andonov on 10/10/16.
//  Copyright © 2016 nikolay.andonov. All rights reserved.
//

#import "SampleTableViewController.h"
#import "CoreDataHandler.h"

@interface SampleTableViewController ()

@property (nonatomic, strong) NSArray<SampleEntity *> *samplesArray;

@end

@implementation SampleTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.tableView.backgroundColor = [UIColor lightGrayColor];
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"Cell"];
    self.tableView.tableFooterView = [[UIView alloc] init];
    
    NSManagedObjectContext *context = [CoreDataStack sharedInstance].persistentContainer.viewContext;
    self.samplesArray = [CoreDataHandler fetchAllSamplesWithContext:context];
}


#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.samplesArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
    SampleEntity *sampleEntity = self.samplesArray[indexPath.row];
    cell.textLabel.text = sampleEntity.name;
    return cell;
}

#pragma mark - Table view delegate

-(void)tableView:(UITableView *)tableView didDeselectRowAtIndexPath:(NSIndexPath *)indexPath {
    
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
}

@end

File: NSPersistentContainer/NSPersistentContainerSwiftSample/NSPersistentContainerSwiftSample/SampleTableViewController.swift
//
//  SampleTableViewController.swift
//  NSPersistentContainerSwiftSample
//
//  Created by nikolay.andonov on 10/10/16.
//  Copyright © 2016 nikolay.andonov. All rights reserved.
//

import UIKit
import CoreData

class SampleTableViewController: UITableViewController {
    
    var samplesArray: NSArray = []
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.tableView.backgroundColor = UIColor.lightGray
        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
        self.tableView.tableFooterView = UIView()
        
        let context : NSManagedObjectContext = CoreDataStack.sharedInstance.persistentContainer.viewContext
        samplesArray = CoreDataHandler.fetchAllSamples(context: context)!
    }
    
    
    // MARK: - Table view data source
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return samplesArray.count
    }
    
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
        let sampleEntity : SampleEntity = samplesArray[indexPath.row] as! SampleEntity
        cell.textLabel?.text = sampleEntity.name
        return cell
    }
    
    // MARK: - Table view delegate
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        tableView.deselectRow(at: indexPath, animated: true)
    }
}
